### PR TITLE
DAOS-16824 cart: lower error messages to warn level

### PR DESCRIPTION
**Pull request by jxiong**
Lower some chatty messages from error level to warn level.

Ideally the error messages should be printed by callers because they should know how serious the error is.

This PR also removes some error messages because they are being printed repeatedly due to the same error.

Signed-off-by: Jinshan Xiong <jinshanx@google.com>

Allow-unstable-test: true
Change-Id: I71966a235c068abbd3c74c422458a68e3a86154a

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/cart/crt_context.c [hunk 1]
```diff
@@ -1256,48 +1254,17 @@ crt_context_timeout_check(struct crt_context *crt_ctx)
...
-
-		} else {
-			RPC_INFO(rpc_priv,
-				 "ctx_id %d, (status: %#x) timed out (%d seconds), "
-				 "target (%d:%d)\n",
-				 crt_ctx->cc_idx, rpc_priv->crp_state, rpc_priv->crp_timeout_sec,
-				 rpc_priv->crp_pub.cr_ep.ep_rank, rpc_priv->crp_pub.cr_ep.ep_tag);
-			left_to_print--;
-		}
+		/* NB: The reason that the error message is printed at INFO
```
Comment (frostedcmos) [inline src/cart/crt_context.c:1260]: in many instances we dont have info level logging on by default. i would prefer the first message to remain as warn, while subsequent ones can be downgraded to info.

File: src/cart/crt_hg.c [hunk 2]
```diff
@@ -1154,9 +1154,9 @@ crt_rpc_handler_common(hg_handle_t hg_hdl)
 	else
 		rc = crt_corpc_common_hdlr(rpc_priv);
 	if (unlikely(rc != 0)) {
-		RPC_ERROR(rpc_priv,
-			  "failed to invoke RPC handler, rc: "DF_RC"\n",
-			  DP_RC(rc));
+		RPC_WARN(rpc_priv,
+			 "failed to invoke RPC handler, rc: "DF_RC"\n",
+			 DP_RC(rc));
```
Comment (frostedcmos) [inline src/cart/crt_hg.c:1159]: i expect either crp_rpc_common_hdlr() or crt_corpc_common_hdlr() to print reason for failure if they return rc != 0, so probably we can just nuke message here altogether

File: src/cart/crt_hg.c [hunk 3]
```diff
@@ -1364,8 +1364,8 @@ crt_hg_req_send_cb(const struct hg_cb_info *hg_cbinfo)
 		crt_cbinfo.cci_rc = rc;
 
 		if (crt_cbinfo.cci_rc != 0)
-			RPC_CERROR(crt_quiet_error(crt_cbinfo.cci_rc), DB_NET, rpc_priv,
-				   "RPC failed; rc: " DF_RC "\n", DP_RC(crt_cbinfo.cci_rc));
+			RPC_WARN(rpc_priv,
+				 "RPC failed; rc: " DF_RC "\n", DP_RC(crt_cbinfo.cci_rc));
```
Comment (frostedcmos) [inline src/cart/crt_hg.c:1368]: we need something like RPC_CWARN here that takes the same condition. you dont want to print warnings for all group version mismatches either, and those are very frequent

File: src/cart/crt_hg_proc.c [hunk 4]
```diff
@@ -712,14 +712,9 @@ crt_proc_out_common(crt_proc_t proc, crt_rpc_output_t *data)
...
-				RPC_ERROR(rpc_priv,
-					  "RPC failed to execute on target. "
-					  "error code: "DF_RC"\n", DP_RC(rc2));
-			else
-				RPC_TRACE(DB_NET, rpc_priv,
-					  "RPC failed to execute on target. "
-					  "error code: "DF_RC"\n", DP_RC(rc2));
+			RPC_WARN(rpc_priv,
+				 "RPC failed to execute on target. "
+				 "error code: "DF_RC"\n", DP_RC(rc2));
```
Comment (frostedcmos) [inline src/cart/crt_hg_proc.c:717]: same as above, you dont really want to print these as warns for DER_GRPVER either

**Review by frostedcmos — COMMENTED** [review]
comments inline

File: src/cart/crt_hg.c (hunk 2 above)
Comment (jxiong) [reply src/cart/crt_hg.c:1159]: Yes, I kept the error message over there. How about we change this to INFO just in case it will miss thing?

File: src/cart/crt_context.c (hunk 1 above)
Comment (jxiong) [reply src/cart/crt_context.c:1260]: I can do that.

File: src/cart/crt_hg.c (hunk 3 above)
Comment (jxiong) [reply src/cart/crt_hg.c:1368]: For sure.

File: src/cart/crt_context.c [hunk 5]
```diff
@@ -534,8 +534,8 @@ crt_rpc_complete_and_unlock(struct crt_rpc_priv *rpc_priv, int rc)
 			cbinfo.cci_rc = rpc_priv->crp_reply_hdr.cch_rc;
 
 		if (cbinfo.cci_rc != 0)
-			RPC_CERROR(crt_quiet_error(cbinfo.cci_rc), DB_NET, rpc_priv,
-				   "failed, " DF_RC "\n", DP_RC(cbinfo.cci_rc));
+			RPC_WARN(rpc_priv,
```
Comment (frostedcmos) [inline src/cart/crt_context.c:537]: this needs to be RPC_CWARN as well

**Review by frostedcmos — COMMENTED** [review]
looks good. 1 more place to change, noted inline

**Review by frostedcmos — APPROVED** [review]
looks good, thanks:)

**Review by jolivier23 — APPROVED** [review]

**Comment by jxiong** [comment]
NLT is broken - I have seen this from multiple of my PRs.

**Comment by mchaarawi** [comment]
i don't have a strong opinion one way or the other for this change.
we do have a lot of chatty error logging in several areas, especially timeouts.
lowering to info does not really do much, because 99% of users actually do use info logging, not ERR or WARN. 

my main concern on this is what if there are actual errors on the fabric (not just timeouts). are these also downgraded to warn / info messages?

**Comment by jxiong** [comment]
> i don't have a strong opinion one way or the other for this change. we do have a lot of chatty error logging in several areas, especially timeouts. lowering to info does not really do much, because 99% of users actually do use info logging, not ERR or WARN.

Well, at least the users would have a choice to adjust the log error. Right now, the rate error message is at ~400 messages per second at error level.

On the server side, one of the major problem is that the same error now is printed thrice, in crt_rpc_handler_common(),  crt_rpc_common_hdlr(), and crt_proc_out_common() respectively. This PR lowers the message level in crt_rpc_handler_common() and crt_proc_out_common() but keeps the error message on crt_rpc_common_hdlr().

> 
> my main concern on this is what if there are actual errors on the fabric (not just timeouts). are these also downgraded to warn / info messages?

This PR is not intended to change any internal errors in fabric. I tend to think the framework should only mind its own internal errors. The users will know how serious an error is then decide to print out something. As my example shows in the ticket, evetually the error will be printed by CLI, why would the framework bother printing those errors? 

Another issue is the same as I mentioned previously - the same error is being printed over and over again in the stack. This PR does its best to remove those duplicated messages.

**Comment by jolivier23** [comment]
> i don't have a strong opinion one way or the other for this change. we do have a lot of chatty error logging in several areas, especially timeouts. lowering to info does not really do much, because 99% of users actually do use info logging, not ERR or WARN.
> 
> my main concern on this is what if there are actual errors on the fabric (not just timeouts). are these also downgraded to warn / info messages?

@mchaarawi our logs explorer allows us to filter messages based on the log level during or after the logs have been produced.  Also, ERROR messages are marked "red" in the timeline and it would be useful to only see red when it's an actual error to make it easier to spot the root cause of real problems without having to sift through noise.  Additionally, there is a psychological issue for customer who sees a bunch of errors vs warnings or informational messages.   If we flag everything as an error, especially when it's not fatal, it gives a worse impression, IMO

**Review by soumagne — DISMISSED** [review]

**Review by jolivier23 — DISMISSED** [review]

**Review by frostedcmos — APPROVED** [review]

**Comment by jolivier23** [comment]
Looks like it hit some NLT issues.  Try merging latest and running with 

Allow-unstable-test: true in final commit message to make sure it runs full tests regardless.

File: src/tests/simple_dfs.c [hunk 6]
```diff
@@ -48,8 +48,16 @@ main(int argc, char **argv)
 	rc = dfs_init();
```
Comment (mchaarawi) [inline src/tests/simple_dfs.c:48]: changes in this file are unrelated to the PR. a wrong merge perhaps?

Comment (jxiong) [reply src/tests/simple_dfs.c:48]: oops

**Review by frostedcmos — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]
